test/fakeintake/client/client.go

```diff
@@ -782,13 +811,35 @@ func (c *Client) get(route string) ([]byte, error) {
 	var body []byte
 	err := backoff.Retry(func() error {
 		tmpResp, err := http.Get(fmt.Sprintf("%s/%s", c.fakeIntakeURL, route))
+		if err, ok := err.(net.Error); ok && err.Timeout() {
+			return ErrFakeintakeTimeout
+		}
 		if err != nil {
 			return err
 		}
+
 		defer tmpResp.Body.Close()
 		if tmpResp.StatusCode != http.StatusOK {
-			return fmt.Errorf("Expected %d got %d", http.StatusOK, tmpResp.StatusCode)
+			return fmt.Errorf("expected %d got %d", http.StatusOK, tmpResp.StatusCode)
 		}
+		// If strictFakeintakeIDCheck is enabled, we check that the fakeintake ID is the same as the one we expect
+		// If the fakeintake ID is not set yet we set the one we get from the first request
+		// If the fakeintake does not return its id in the header we do not check it
+		if c.strictFakeintakeIDCheck && tmpResp.Header.Get(fakeintakeIDHeader) != "" {
+			if c.fakeintakeID == "" {
+				c.fakeintakeIDMutex.Lock()
+				c.fakeintakeID = tmpResp.Header.Get(fakeintakeIDHeader)
+				c.fakeintakeIDMutex.Unlock()
+			} else {
+				c.fakeintakeIDMutex.RLock()
+				currentFakeintakeID := c.fakeintakeID
+				c.fakeintakeIDMutex.RUnlock()
+				if currentFakeintakeID != tmpResp.Header.Get(fakeintakeIDHeader) {
+					return fmt.Errorf("expected %s got %s: %w", currentFakeintakeID, tmpResp.Header.Get(fakeintakeIDHeader), ErrFakeintakeRestarted)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I would panic here, so that any calls panics as well. If we return an error then we need to handle it in the caller, it is safe to handle it here

Reply:
That sounds good. I need to investigate a bit how I can unit test code that should panic